### PR TITLE
Higher jump

### PR DESCRIPTION
**Pull request by w1158mxs**
fix Issue#39 part of Issue#70 Issue#71
-Players can make higher when they speed and lower when decreasing speed with the same key down Q & E
-Link Issue numbers for some screens.
-Correct Mistakes in the Read Me file.

Canceled Issue#33 and Issue#34 because of conflict with Issue#42 which its blocks the user to build anything on the cloud or remove them - player now can't build clouds.

### CODE REVIEW

**Review by Lurk390 — APPROVED** [review]
Code looks good to me, however I still feel a little iffy about this feature as a whole. I'll approve it but I might tweak it a bit in a future pr.

File: tempus_fugit_minecraft/model.py
Comment (xieWright) [inline tempus_fugit_minecraft/model.py:553]: Needs to change to be more specific. With how it is, if increase is anything that's not False or 0, it will always default to the first if statement.

File: tests/test_jump
Comment (xieWright) [inline tests/test_jump:1]: Add a test for handle_jump_change, increase_jump and decrease_jump to test if the functions are working.

File: tempus_fugit_minecraft/player.py
Comment (xieWright) [inline tempus_fugit_minecraft/player.py:119]: Feels like these functions should be relative to the current walking speed rather than the jump_speed. If you want to limit the jump_speed, I think a hard cap would be added, such as max_jump_speed, min_jump_speed.

**Review by xieWright — CHANGES_REQUESTED** [review]
Some nitpicks here and there. The change to increase_jump and decrease_jump are arguable.

File: tempus_fugit_minecraft/player.py
Comment (w1158mxs) [reply tempus_fugit_minecraft/player.py]: Good one, I added 2 variables for max and min, Thank you

File: tempus_fugit_minecraft/model.py
Comment (w1158mxs) [reply tempus_fugit_minecraft/model.py]: Return None, updated! Thank you!

File: tests/test_jump
Comment (w1158mxs) [reply tests/test_jump:1]: I did a jump speed test! after your urgent merge for oxygen, I"ll do the next testes.

**Review by xieWright — CHANGES_REQUESTED** [review]
You still need to change the handle_jump function before I can approve. In addition, the test file needs to be updated to include that function since it's what handles increasing and decreasing jump speed.

File: tempus_fugit_minecraft/player.py
Comment (w1158mxs) [reply tempus_fugit_minecraft/player.py]: Updated as bool

File: tests/test_jump
Comment (criticalglitch) [inline tests/test_jump:1]: Can you combine this new TestJump, TestPauseMenu, TestSpeed, and TestFlying into a single TestWindow class? That way we only spin up 1 window for all of those tests instead of 4 windows. This will cause the tests to run faster.

File: tempus_fugit_minecraft/player.py
Comment (criticalglitch) [inline tempus_fugit_minecraft/player.py:148]: Since we're modifying the jump speed, maybe we should call these `increase_jump_speed` and `decrease_jump_speed`.

File: tempus_fugit_minecraft/player.py
Comment (criticalglitch) [inline tempus_fugit_minecraft/player.py:118]: Similar to above since this is getting renamed, `increase_walk_speed`

File: tempus_fugit_minecraft/player.py
Comment (criticalglitch) [inline tempus_fugit_minecraft/player.py:130]: Similar to above, `decrease_walk_speed`

File: tests/test_jump
Comment (criticalglitch) [inline tests/test_jump:27]: These tests will fail currently. Right now the player jump_speed is set to be the sqrt(40), (more specifically:`sqrt(2 * GRAVITY * MAX_JUMP_HEIGHT) == sqrt(2 * 20 * 1) == sqrt(40)`) which is roughly 6. So, adding 5 to these won't make 10, 15 or 20, it will make 11, 16, and 21 (also roughly).

Also, these tests don't actually run since this isn't a `.py` file.

File: tempus_fugit_minecraft/window.py
Comment (criticalglitch) [inline tempus_fugit_minecraft/window.py:328]: Docstrings go inside the functions, not above them. See here: [PEP-257](https://peps.python.org/pep-0257/)

This applies to all other similar changes, such as for `set_3d`, `on_draw`, `draw_pause_menu`, `draw_focused_block`, `draw_reticle`, `decrease_speed`, and `increase_speed`.

File: tempus_fugit_minecraft/player.py
Comment (criticalglitch) [reply tempus_fugit_minecraft/player.py:148]: Also, please add doxygen style comments similar to @bakursait2's PR

**Review by xieWright — CHANGES_REQUESTED** [review]
Add a test for the handle_jump_change since it's what chooses between the options.

File: tempus_fugit_minecraft/model.py
Comment (criticalglitch) [inline tempus_fugit_minecraft/model.py:532]: Please link the issue in the doxygen comment like how @bakursait2's PR has it:
`@see [Issue 39](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/39)`

**Review by criticalglitch — CHANGES_REQUESTED** [review]
The changes look alright, but I think some changes need to be made before we can merge this at this point.

File: tests/test_jump
Comment (xieWright) [reply tests/test_jump:1]: I would say that TestJump and TestSpeed should be put into the testPlayer file since these affect properties of the player and contain functions of the player. But I think this is a minor issue that can be addressed in another pull request. Though I don't think it will count as a significant pull request.

Comment (criticalglitch) [reply tests/test_jump:1]: TestSpeed and TestJump are testing that triggering certain inputs on a keyboard triggers the desired effects of the player walk or jump speed changing.  The thing under test in those two classes *is the Window*, not the player, because we are calling functions *on the Window*. The fact that we're asserting stuff on the player is only relevant because of what the test is dictating. Specifically: "if I call Window.on_key_pressed and pass it a Q, the speed of the player in the model of the window should increase." 


So we really should combine those four test classes because they're all testing parts of the Window.

**Review by bakursait2 — APPROVED** [review]
The Tests pass and the game works.
Also consider `rebase` on top of the master to get the latest updates and to avoid any conflicts.

**Comment by w1158mxs** [comment]
I have a better idea to program it and make it clean code I'm working on it, I'll try it soon!

File: tests/test_jump
Comment (w1158mxs) [reply tests/test_jump:1]: > Can you combine this new TestJump, TestPauseMenu, TestSpeed, and TestFlying into a single TestWindow class? That way we only spin up 1 window for all of those tests instead of 4 windows. This will cause the tests to run faster.

can anyone collect them in a new issue if we have more time?

File: tempus_fugit_minecraft/player.py
Comment (w1158mxs) [reply tempus_fugit_minecraft/player.py]: > Also, please add doxygen style comments similar to @bakursait2's PR

Done!

File: tempus_fugit_minecraft/player.py
Comment (w1158mxs) [reply tempus_fugit_minecraft/player.py:118]: > Similar to above since this is getting renamed, `increase_walk_speed`

Done!

File: tempus_fugit_minecraft/player.py
Comment (w1158mxs) [reply tempus_fugit_minecraft/player.py:128]: > Similar to above, `decrease_walk_speed`

Done!

File: tests/test_jump
Comment (w1158mxs) [reply tests/test_jump]: > These tests will fail currently. Right now the player jump_speed is set to be the sqrt(40), (more specifically:`sqrt(2 * GRAVITY * MAX_JUMP_HEIGHT) == sqrt(2 * 20 * 1) == sqrt(40)`) which is roughly 6. So, adding 5 to these won't make 10, 15 or 20, it will make 11, 16, and 21 (also roughly).
> 
> Also, these tests don't actually run since this isn't a `.py` file.

file Fixed with .py extension, and test pass, Done!

File: tempus_fugit_minecraft/window.py
Comment (w1158mxs) [reply tempus_fugit_minecraft/window.py]: Done

File: tempus_fugit_minecraft/model.py
Comment (xieWright) [inline tempus_fugit_minecraft/model.py:496]: Just an editing change.

**Review by xieWright — CHANGES_REQUESTED** [review]
Just a small editing change, but other than that should be fine.

File: tempus_fugit_minecraft/model.py
Comment (criticalglitch) [inline tempus_fugit_minecraft/model.py]: Like above, This PR is blocked until this merge conflict is resolved. We need to accept the current change for this diff.

**Review by criticalglitch — CHANGES_REQUESTED** [review]

**Review by criticalglitch — APPROVED** [review]

**Review by xieWright — APPROVED** [review]
Seems done now. No issues with the check.

**Review by bakursait2 — APPROVED** [review]
The code works fine and the tests passed. The feature is a new addition to the game.

Thank you.

**Pull request merged**
